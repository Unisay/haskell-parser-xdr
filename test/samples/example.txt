[ Right (ConstantDef (Identifier "MAXUSERNAME") (DecConstant 32))
, Right (ConstantDef (Identifier "MAXFILELEN") (DecConstant 65535))
, Right (ConstantDef (Identifier "MAXNAMELEN") (DecConstant 255))
, Left
    (TypeDefEnum
       (Identifier "filekind")
       (( Identifier "TEXT" , Left (DecConstant 0) ) :|
          [ ( Identifier "DATA" , Left (DecConstant 1) )
          , ( Identifier "EXEC" , Left (DecConstant 2) )
          ]))
]