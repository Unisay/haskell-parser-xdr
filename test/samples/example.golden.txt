[ Right (ConstantDef (Identifier "MAXUSERNAME") (DecConstant 32))
, Right (ConstantDef (Identifier "MAXFILELEN") (DecConstant 65535))
, Right (ConstantDef (Identifier "MAXNAMELEN") (DecConstant 255))
, Left
    (TypeDefEnum
       (Identifier "filekind")
       (( Identifier "TEXT" , Left (DecConstant 0) ) :|
          [ ( Identifier "DATA" , Left (DecConstant 1) )
          , ( Identifier "EXEC" , Left (DecConstant 2) )
          ]))
, Left
    (TypeDefUnion
       (Identifier "filetype")
       UnionBody
         { unionDiscriminant =
             DeclarationSingle
               (TypeIdentifier (IdentifierRef "filekind")) (Identifier "kind")
         , unionArms =
             CaseSpec
               { caseSpecValues = Right (IdentifierRef "TEXT") :| []
               , caseSpecDeclaration = DeclarationVoid
               } :|
               [ CaseSpec
                   { caseSpecValues = Right (IdentifierRef "DATA") :| []
                   , caseSpecDeclaration =
                       DeclarationString
                         (Identifier "creator") (Just (Right (IdentifierRef "MAXNAMELEN")))
                   }
               , CaseSpec
                   { caseSpecValues = Right (IdentifierRef "EXEC") :| []
                   , caseSpecDeclaration =
                       DeclarationString
                         (Identifier "interpretor") (Just (Right (IdentifierRef "MAXNAMELEN")))
                   }
               ]
         , unionDefault =
             Just
               (DeclarationSingle
                  (TypeIdentifier (Identifier "foo")) (Identifier "bar"))
         })
, Left
    (TypeDefStruct
       (Identifier "file")
       (DeclarationString
          (Identifier "filename") (Just (Right (Identifier "MAXNAMELEN"))) :|
          [ DeclarationSingle
              (TypeIdentifier (Identifier "filetype")) (Identifier "type")
          , DeclarationString
              (Identifier "owner") (Just (Right (Identifier "MAXUSERNAME")))
          , DeclarationOpaqueVarLen
              (Identifier "data") (Just (Right (Identifier "MAXFILELEN")))
          ]))
]
